### PR TITLE
[v14] Allow configurable Okta service synchronization duration.

### PR DESCRIPTION
**Pull request by mdwn**
Backport #31170 to branch/v14

### CODE REVIEW

**Review by avatus — APPROVED** [review]

**Review by fheinecke — APPROVED** [review]
~~Merging as this appears to be holding up the v14 release~~ Need a G2 approval

ref: https://github.com/gravitational/teleport/actions/runs/6029682353/job/16359781041?pr=31253

**Review by r0mant — APPROVED** [review]

**Review by AntonAM — APPROVED** [review]

**Pull request merged**
